src/orquestra/sdk/_base/cli/_dorq/_repos.py

```diff
@@ -133,6 +159,88 @@ def get_wf_outputs(self, wf_run_id: WorkflowRunId, config_name: ConfigName):
 
         return outputs
 
+    def _get_wf_def_model(
+        self, wf_run_id: WorkflowRunId, config_name: ConfigName
+    ) -> WorkflowDef:
+        """
+        Raises:
+            orquestra.sdk.exceptions.NotFoundError: when the wf_run_id doesn't match a
+                stored run ID.
+            orquestra.sdk.exceptions.ConfigNameNotFoundError: when the named config is
+                not found in the file.
+        """
+        try:
+            wf_run = sdk.WorkflowRun.by_id(wf_run_id, config_name)
+        except (exceptions.NotFoundError, exceptions.ConfigNameNotFoundError):
+            raise
+
+        status_model = wf_run.get_status_model()
+        wf_def = status_model.workflow_def
+
+        # TODO: check when this is optional and add a message
+        assert wf_def is not None
+
+        return wf_def
+
+    def get_task_fn_names(
+        self, wf_run_id: WorkflowRunId, config_name: ConfigName
+    ) -> t.Sequence[str]:
+        """
+        Extracts task function names used in this workflow run.
+
+        If two different task defs have the same function name this returns a single
+        name entry. This can happen when similar tasks are defined in two different
+        modules.
+
+        Raises:
+            orquestra.sdk.exceptions.NotFoundError: when the wf_run_id doesn't match a
+                stored run ID.
+            orquestra.sdk.exceptions.ConfigNameNotFoundError: when the named config is
+                not found in the file.
+        """
+
+        try:
+            wf_def = self._get_wf_def_model(wf_run_id, config_name)
+        except (exceptions.NotFoundError, exceptions.ConfigNameNotFoundError):
+            raise
+
+        # What we're really interested in are task_runs, but it's easier to test when
+        # we iterate over wf_defs's task_defs. Assumption: every task_def is being used
+        # in the workflow and corresponds to a task_run.
+        names_set = {
+            task_def.fn_ref.function_name for task_def in wf_def.tasks.values()
```

Comment:
Nonblocking: thought: Is the function_name enough of an identifier? What if we have 2 functions in 2 different files/modules with exact same name. How would user distinguish between them? Or even 2 methods in 2 different classes

Reply:
Btw. This is nonblocking for this PR, but still requires analysis. If you consider it out of scope, lets make Jira ticket for that

Reply:
It's a known limitation described in the docstring for `get_task_fn_names`:
```
        Extracts task function names used in this workflow run.

        If two different task defs have the same function name this returns a single
        name entry. This can happen when similar tasks are defined in two different
        modules.
```

I made an attempt to use a different way to identify the task defs to avoid shadowing (see af7d9c952a306cd313003fdc330cff7749899b4c and the revert coming right after it), and it's possible, but I couldn't figure out a user-friendly way of providing this as an CLI arg. Asking the user to provide a `task_def_id` would suck, defeating the whole purpose of these filters

Reply:
> Is the function_name enough of an identifier?

There are situations where it's not, and that's why there's another inquiry step for disambiguating between task invocations :)